### PR TITLE
fix delay & onqueue issues

### PR DESCRIPTION
**Pull request by deviouspk**
Please note that i did NOT test this yet. But should work nonetheless

### CODE REVIEW

**Comment by deviouspk** [comment]
Fix for this issue: https://github.com/apiato/apiato/issues/455

**Comment by deviouspk** [comment]
fixes all problems. Tested & fully working

**Comment by deviouspk** [comment]
this feature was kind of broken. so we might want to release this asap.

File: Abstracts/Events/Dispatcher/Dispatcher.php [hunk 1]
```diff
@@ -14,18 +14,34 @@
...
-            $job = new EventJob($event);
-            $job = isset($event->jobDelay) ? $job->delay($event->jobDelay) : $job;
-            $job = isset($event->jobQueue) ? $job->onQueue($event->jobQueue) : $job;
-            new JobDispatcher($job);
-        } else if ($event instanceof ShouldHandleNow) {
-            $event->handle();
-        }
-
-        return parent::dispatch($event, $payload, $halt);
+  public function dispatch($event, $payload = [], $halt = false)
```
Comment (johannesschobel) [inline Abstracts/Events/Dispatcher/Dispatcher.php]: can you please fix indentation from 2 spaces to 4 spaces throughout your changes? thank you ;)

File: Abstracts/Events/Interfaces/ShouldHandleNow.php [hunk 2]
```diff
@@ -7,7 +7,7 @@
  *
  * @author  Arthur Devious
  */
-interface ShouldHandleNow extends ShouldHandle
+interface ShouldHandleNow
```
Comment (johannesschobel) [inline Abstracts/Events/Interfaces/ShouldHandleNow.php]: What is your opinion on this:
```
interface ShouldHandleEvents
interface ShouldHandleEventsImmediately extends ShouldHandleEvents
interface ShouldHandleEventsQueued extends ShouldHandleEvents
```

Would that be a `Breaking Change`?

Comment (deviouspk) [reply Abstracts/Events/Interfaces/ShouldHandleNow.php]: well people who have implemented the old interfaces will have to adjust their code. So when they update apiato to the latest version their code won't work anymore until they've migrated to the new interfaces

**Comment by deviouspk** [comment]
Formatted the code

File: Abstracts/Events/Interfaces/ShouldHandleNow.php (hunk 2 above)
Comment (johannesschobel) [reply Abstracts/Events/Interfaces/ShouldHandleNow.php]: that is true, yes.. hm - you're right ;)

Comment (deviouspk) [reply Abstracts/Events/Interfaces/ShouldHandleNow.php]: we can deprecate if you want but i think the naming is fine :p can you merge the pull request?

Comment (deviouspk) [reply Abstracts/Events/Interfaces/ShouldHandleNow.php]: johannes would you be so kind to merge this pull request to the master :p and release a new version ?

**Comment by johannesschobel** [comment]
i will merged it for now.. i haven't had the time to check it, though.. :(

**Pull request merged**